### PR TITLE
feat(185): Add extraction page with part 1 of integration

### PR DESCRIPTION
**Pull request by knguyenrise8**
# Pull Request

github issue 185

part 1 of pr
loading spinner and api integration without, add cors

next part for part 2 will include 

PDF image below and cors integration

## Description
[Provide a brief description of the changes implemented in this pull request.]

## Related Issues
[Link any related issues or tasks from your project management system.]

## Additional Notes
[Add any additional context or notes that reviewers should know about.]

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.
- [ ] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

**Review by schreiaj — APPROVED** [review]

File: OCR/frontend/api/api.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    form.append("segmentation_template", templateImage);
+    form.append("labels", JSON.stringify(fieldNames));
+    form.append("", "");
+    console.log(args)
+    try {
+        const response = await fetch("http://localhost:8000/image_to_text/", {
+            "method": "POST",
+            "headers": {
+              "Content-Type": "multipart/form-data",
+              "content-type": "multipart/form-data; boundary=---011000010111000001101001"
```
Comment (zdeveloper) [inline OCR/frontend/api/api.ts:19]: is the boundary automatically generated? or is it hardcoded?

File: OCR/frontend/src/components/ImageAnnotator.tsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ export const MultiImageAnnotator: FC<MultiImageAnnotatorProps> = ({ images, cate
         const field = fields.find(field => field.name === selectedField?.name);
         const updatedShapes = [...shapes];
         // todo fix typing but for now this is fine
-        updatedShapes[index] = [...(updatedShapes[index] || []), {...shape, field: selectedField?.name as string, color: field?.color}];
+        updatedShapes[index] = [...(updatedShapes[index] || []), {...shape, field: selectedField?.name as string, color: field?.color.slice(0,7)}];
```
Comment (zdeveloper) [inline OCR/frontend/src/components/ImageAnnotator.tsx:38]: getting this error on this line
image

File: OCR/frontend/src/pages/SaveTemplate.tsx [hunk 3]
```diff
@@ -1,43 +1,52 @@
...
+import { Label, TextInput } from "@trussworks/react-uswds";
+import { Divider } from "../components/Divider";
+import { UploadHeader } from "../components/Header";
+import { Stepper } from "../components/Stepper";
+import { AnnotateStep } from "../utils/constants";
+import { useNavigate } from "react-router-dom";
+import { useAnnotationContext } from "../contexts/AnnotationContext";
+import { useFiles, FileType, Page } from "../contexts/FilesContext";
+import hexRgb from "hex-rgb";
+import { base64ToBinaryFile } from "../utils/utils";
```
Comment (zdeveloper) [inline OCR/frontend/src/pages/SaveTemplate.tsx:10]: This is not used, I am assuming it will be used in part 2

**Review by zdeveloper — APPROVED** [review]
LGTM for part 1

File: OCR/frontend/src/pages/SaveTemplate.tsx (hunk 3 above)
Comment (knguyenrise8) [reply OCR/frontend/src/pages/SaveTemplate.tsx:10]: yes

**Review by zdeveloper — APPROVED** [review]

**Pull request merged**